### PR TITLE
Remove nginx container from mc gateway pod

### LINKED ISSUE #5444: Remove nginx container from mc gateway pod
The multicluster `gateway` component includes an `nginx` container; but this container is not actually used at all. (Instead, the proxy routes traffic internally). We should remove this container in favor of a `pause` container.

My one concern is that the service-mirror healthcheck may rely on this container to prove the gateway is live; but this is really checking something unrelated to the functionality of the gateway. Can we replace these healthchecks with requests that actually hit the proxy's gateway service?

### PR DESCRIPTION
**Pull request by alpeb**
Fixes #5444

Replaced the nginx container with a pause container, and removed all the
config boilerplate associated to nginx.

The probe now targets the proxy's readiness endpoint, port 4191 path
`/ready`.

## Upgrade note

When multicluster gets updated in both source and target this won't
cause downtime because this change only affects the probing. The `Links`
will have to be re-generated of course. Til that is done, `linkerd mc gateways`
will show the target cluster as not alive, but that shouldn't affect exiting endpoints
pointing to the target cluster.

### CODE REVIEW

**Review by adleong — APPROVED** [review]

File: multicluster/charts/linkerd-multicluster/templates/gateway.yaml [hunk 1]
```diff
@@ -72,44 +24,14 @@ spec:
...
     spec:
-      volumes:
-        - name: config
-          configMap:
-            name: {{.Values.gatewayName}}-config
-        - name: tmp
-          emptyDir: {}
       containers:
-        - name: nginx
-          readinessProbe:
```
Comment (Pothulapati) [inline multicluster/charts/linkerd-multicluster/templates/gateway.yaml:88]: Should'nt we add these `readinessProbes` to a locally known proxy endpoint or is it not necessary? 🤔

**Review by Pothulapati — COMMENTED** [review]
LGTM, Nice that we also now don't display these extra varialbes to user any more i.e localprobe, etc!

Can you also [update the testutil code to remove the nginx container name.](https://github.com/linkerd/linkerd2/blob/alpeb/bye-bye-nginx/testutil/test_helper.go#L141-L142) Also, Wondering why it did not fail. Are we not checking for containers in `TestResourcesPostInstall`

File: multicluster/charts/linkerd-multicluster/templates/gateway.yaml (hunk 1 above)
Comment (alpeb) [reply multicluster/charts/linkerd-multicluster/templates/gateway.yaml:88]: The `pause` container is mostly a placeholder, no need to setup probes, IMO.

**Comment by alpeb** [comment]
> Can you also [update the testutil code to remove the nginx container name.](https://github.com/linkerd/linkerd2/blob/alpeb/bye-bye-nginx/testutil/test_helper.go#L141-L142) Also, Wondering why it did not fail. Are we not checking for containers in `TestResourcesPostInstall`

Good catch, I've updated that. It seems thought we're indeed not checking those containers. I'll address that in a separate PR.

**Comment by alpeb** [comment]
The `cni-calico-deep` test is failing because quay.io is down. Will try a rerun later :pray:

**Review by Pothulapati — APPROVED** [review]
LGTM

**Pull request merged**